xdsl/transforms/csl_stencil_to_csl_wrapper.py

```diff
@@ -0,0 +1,232 @@
+from collections.abc import Sequence
+from dataclasses import dataclass
+
+from xdsl.builder import ImplicitBuilder
+from xdsl.context import MLContext
+from xdsl.dialects import arith, builtin, func, stencil
+from xdsl.dialects.builtin import IntegerAttr, TensorType
+from xdsl.dialects.csl import csl, csl_stencil, csl_wrapper
+from xdsl.ir import Attribute
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.rewriter import InsertPoint
+from xdsl.utils.hints import isa
+
+
+@dataclass(frozen=True)
+class ConvertStencilFuncToModuleWrappedPattern(RewritePattern):
+    """
+    Wraps program in the csl_stencil dialect in a csl_wrapper module.
+    Scans the csl_stencil.apply ops for stencil-related params, passing them as properties to the wrapped module
+    (note, properties are in return passed as block args to the layout_module and program_module blocks).
+
+    The layout module wrapper can be used to initialise general program module params. This pass generates code
+    to initialise stencil-specific program params and yields them from the layout module.
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: func.FuncOp, rewriter: PatternRewriter, /):
+        # find csl_stencil.apply ops, abort if there are none
+        apply_ops = self.get_csl_stencil_apply_ops(op)
+        if len(apply_ops) == 0:
+            return
+        neighbours: int = 1
+        width: int = 1
+        height: int = 1
+        z_dim_no_ghost_cells: int = 1
+        z_dim: int = 1
```

Comment:
Would love to see a quick comment on these explaining what they are tracking (especially `neighbors`, as I suspect this is in actuality `max_radius`?).